sources/models/src/modeled_types/credential_providers.rs

```diff
@@ -0,0 +1,128 @@
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+// Just need serde's Error in scope to get its trait methods
+use super::error;
+use serde::de::Error as _;
+use snafu::ensure;
+use std::borrow::Borrow;
+use std::convert::TryFrom;
+use std::fmt;
+use std::ops::Deref;
+
+use crate::SingleLineString;
+
+/// CacheDurationValue is a string that represents the duration the plugin will
+/// cache credentials in-memory if a cache duration is not provided in the
+/// plugin response.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct CacheDurationValue {
+    inner: String,
+}
+
+lazy_static! {
+    // A duration string is a sequence of decimal numbers, each with a
+    // unit suffix, such as "300ms" or "2h45m".
+    // Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+    // Based on restraints from Go's `ParseDuration` funciton.
+    pub(crate) static ref DURATION_FORMAT: Regex = Regex::new(r"^([0-9]*[a-z]+)+$").unwrap();
```

Comment:
If we're constraining this string, does it makes sense to require the correct suffix? If I read this right, this is valid: `000xyz`.

Reply:
Probably, but now based on the plugin name validation I'm wondering if we should be doing any of this validation at all, or if we should just pass along whatever string to the kubelet settings to be validated there.

Reply:
Dropping this in favor of the existing `KubernetesDurationValue` which appears to be the same thing.